packages/nodes-base/nodes/MessageBird/MessageBird.node.ts

```diff
@@ -0,0 +1,336 @@
+					},
+					//hash
+					{
+						displayName: 'Type Details',
+						name: 'typeDetails',
+						type: 'string',
+						default: '',
+						description:
+							'A hash with extra information.<br /> Is only used when a binary message is sent.'
+					},
+
+					{
+						displayName: 'Datacoding',
+						name: 'datacoding',
+						type: 'string',
+						default: '',
+						description:
+							'Using unicode will limit the maximum number of characters to 70 instead of 160'
+					},
+					{
+						displayName: 'Mclass',
+						name: 'mclass',
+						type: 'number',
+						placeholder: 'permissible values from 0-3',
+						typeOptions: {
+							minValue: 0,
+							maxValue: 3
+						},
+						default: '',
+						description:
+							'Indicated the message type. 1 is a normal message, 0 is a flash message.'
+					},
+					//date-time format
+					{
+						displayName: 'Scheduled Date-time',
+						name: 'scheduledDatetime',
+						type: 'string',
+						default: '',
+						placeholder: '2011-08-30T09:30:16.768-04:00',
+						description:
+							'The scheduled date and time of the message in RFC3339 format (Y-m-dTH:i:sP).'
+					},
+					//date-time format
+					{
+						displayName: 'Created Date-time',
+						name: 'createdDatetime',
+						type: 'string',
+						placeholder: '2011-08-30T09:30:16.768-04:00',
+						default: '',
+						description:
+							'The date and time of the creation of the message in RFC3339 format (Y-m-dTH:i:sP).'
+					}
+				]
+			}
+		]
+	};
+
+	async execute(this: IExecuteFunctions): Promise<INodeExecutionData[][]> {
+		const items = this.getInputData();
+		const returnData: IDataObject[] = [];
+
+		let operation: string;
+		let resource: string;
+
+		// For POST
+		let Body: IDataObject;
+		// For Query string
+		let qs: IDataObject;
+
+		let requestMethod: string;
+
+		for (let i = 0; i < items.length; i++) {
+			qs = {};
+
+			resource = this.getNodeParameter('resource', i) as string;
+			operation = this.getNodeParameter('operation', i) as string;
+
+			if (resource === 'sms') {
+				if (operation === 'send') {
+					// ----------------------------------
+					//         sms:send
+					// ----------------------------------
+
+					requestMethod = 'POST';
+					const originator = this.getNodeParameter('originator', i) as string;
+					const body = this.getNodeParameter('message', i) as string;
+
+					Body = {
+						recipients: [],
+						originator,
+						body
+					};
+					const additionalFields = this.getNodeParameter(
+						'additionalFields',
+						i
+					) as IDataObject;
+
+					if (additionalFields.groupIds) {
+						Body.groupIds = additionalFields.groupIds as string;
+					}
+					if (additionalFields.type) {
+						Body.type = additionalFields.type as string;
+					}
+					if (additionalFields.reference) {
+						Body.reference = additionalFields.reference as string;
+					}
+					if (additionalFields.reportUrl) {
+						Body.reportUrl = additionalFields.reportUrl as string;
+					}
```

Comment:
If we only need to transfer properties from `additionalFields` to `Body`, then can we refactor all this to `const secondObj = {...firstObj};`? Also I meant to select the entire `if` series, but the UI apparently doesn't allow to scroll while selecting.